packages/docusaurus-theme-classic/src/themeConfigSchema.js

```diff
@@ -102,6 +102,15 @@ const ColorModeSchema = Joi.object({
   defaultMode: Joi.string().equal('dark', 'light').default('light'),
   disableSwitch: Joi.bool().default(false),
   respectPrefersColorScheme: Joi.bool().default(false),
+  switchConfig: Joi.object({
+    darkIcon: Joi.string().default('🌜'),
+    darkIconStyle: Joi.object(),
+    lightIcon: Joi.string().default('🌞'),
+    lightIconStyle: Joi.object(),
+  }).default({
+    darkIcon: '🌜',
+    lightIcon: '🌞',
+  }),
 }).default({
   defaultMode: 'light',
   disableSwitch: false,
```

Comment:
The e2e tests are failing

> (undefined) TypeError: Cannot destructure property `darkIcon` of 'undefined' or 'null'.



This is because your schema does not work if colorMode config is simply undefined. You need to include a `switchConfig` in the default object as well.

(As we have seen, Joi is not so idiomatic so I'd appreciate if you add validation for those edge cases in the configValidation.test.ts)

Reply:
@Drewbi the expected change is here, not in the component.

```
.default({
  defaultMode: 'light',
  disableSwitch: false,
  respectPrefersColorScheme: false,
  switchConfig: {
    darkIcon: '🌜',
    lightIcon: '🌞',
  },
```

This is weird to have to declare twice these  defaults for switchconfig but this is how Joi works, as it considers undefined and {} differently. @anshulrgoyal do you think there's a better way with Joi? (maybe an option I don't know)